### ISSUE TITLE
the background in alien hominid is black

### ISSUE THREAD

**Issue by dryluggage** [ruffle-rs/ruffle#6365]
### Describe the bug

![bug](https://user-images.githubusercontent.com/58333865/155894056-c02749ec-2a30-40ac-a7b7-3a127e34ce73.png)

### Expected behavior

background shouldnt be black

### Affected platform

Online demo

### Operating system

Arch Linux 5.16.11-arch1-1

### Browser

mozilla firefox 97.0.1

**Comment by dryluggage** [comment]
my graphics card is Intel(R) Q45/Q43

**Comment by relrelb** [comment]
Thanks for the report!
I suspect this is a WebGL limitation. Alien Hominid's background image is pretty large and is usually rendered black on mobile devices that have `MAX_TEXTURE_SIZE` of 4096. Can you please verify it with:
```js
const gl = document.createElement("canvas").getContext("webgl2");
gl.getParameter(gl.MAX_TEXTURE_SIZE);
```
Also, can you please confirm this doesn't happen on desktop nor canvas backend (it can be forced by turning `webgl.disabled` on in Firefox's `about:config`)?

**Comment by dryluggage** [comment]
so i did change webgl.disabled in about config and the background image is working

but i dont quite understand what you want me to do with:
const gl = document.createElement("canvas").getContext("webgl2");
gl.getParameter(gl.MAX_TEXTURE_SIZE);

if you want me to run this in the browserconsole then there is an error in red:

Uncaught TypeError: gl is null
    <anonymous> debugger eval code:2

here is the log before i executed any code this also maybe helpful:

Failed to create WebGL context: WebGL creation failed: 
* tryNativeGL (FEATURE_FAILURE_EGL_NO_CONFIG)
* Exhausted GL driver options. (FEATURE_FAILURE_WEBGL_EXHAUSTED_DRIVERS)

New Ruffle instance created (WebAssembly extensions: ON) ruffle-player.js:250:16

Loading SWF file swfs/alien_hominid.swf ruffle-player.js:370:24

WebGL warning: texImage: Requested size at this level is 
unsupported. ruffle_web-wasm_extensions.js:1237:24

Die fixierte Bildlauf-Position (Scroll Anchoring) wurde in einem Scroll-Container deaktiviert, weil zu viele aufeinanderfolgende Korrekturen (10) mit zu wenig Abstand (-0.166666662693024 Pixel im Durchschnitt, -1.66667 Pixel insgesamt) vorgenommen wurden. demo

Call: Invalid call 2 ruffle_web-wasm_extensions.js:810:16

**Comment by relrelb** [comment]
Yeah, I meant you to run this in the browser console. Strange it didn't work on Firefox (it works on Chrome). Maybe the canvas should be appended to the DOM:
```js
const canvas = document.createElement("canvas");
document.body.appendChild(canvas);
const gl = canvas.getContext("webgl2");
gl.getParameter(gl.MAX_TEXTURE_SIZE);
```

**Comment by dryluggage** [comment]
here is the output:

Failed to create WebGL context: WebGL creation failed: 
* tryNativeGL (FEATURE_FAILURE_EGL_NO_CONFIG)
* Exhausted GL driver options. (FEATURE_FAILURE_WEBGL_EXHAUSTED_DRIVERS) debugger eval code:3:19


maybe a heads up but i dont think my gpu support webgl2

**Comment by relrelb** [comment]
Well, Ruffle can fallback to WebGL1 if WebGL2 is not supported. Try replacing `"webgl2"` with `"webgl"`, and make sure that `webgl.disabled` is turned off.

**Comment by dryluggage** [comment]
i changed "webgl2" to "webgl" and the output was : 4096

**Comment by waspennator** [comment]
I tried it out on NG on my S22 and i'm getting 2 specific messages popping up in the console log. The first one I see is

ruffle_web-wasm_extensions.js:1094 WebGL: INVALID_VALUE: texImage2D: width or height out of range

then as soon as I start the game it starts flooding the log with this.

[.WebGL-0x2035adb00]RENDER WARNING: texture bound to texture unit 0 is not renderable. It might be non-power-of-2 or have incompatible texture filtering (maybe)?

I can see this error message also appear on another animation I tried on my S22
https://www.newgrounds.com/portal/view/512343/format/flash
